pkg/controller/workload/job/job_controller.go

```diff
@@ -219,11 +246,33 @@ func (r *JobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, err
 	}
 
-	// 4.4 workload is admitted and job is running, nothing to do.
+	// workload is admitted and job is running, nothing to do.
 	log.V(3).Info("Job running with admitted workload, nothing to do")
 	return ctrl.Result{}, nil
 }
 
+func jobPodsReady(job *batchv1.Job) bool {
+	return job.Status.Succeeded+podsReadyCount(&job.Status) >= jobPodsReadyExpectedCount(&job.Spec)
+}
+
+func jobPodsReadyExpectedCount(jobSpec *batchv1.JobSpec) int32 {
+	parallelism := *(jobSpec.Parallelism) // parallelism is always set as it is otherwise defaulted by k8s to 1
+	if jobSpec.Completions != nil {
+		if parallelism <= *jobSpec.Completions {
+			return parallelism
+		}
+		return *jobSpec.Completions
+	}
+	return parallelism
+}
+
+func podsReadyCount(jobStatus *batchv1.JobStatus) int32 {
```

Comment:
You can replace this with `pointer.Int32Deref` https://github.com/kubernetes/utils/blob/master/pointer/pointer.go#L86

Reply:
I see, nice

Reply:
too many short functions :) I would remove this function and just use `Int32Deref` directly in jobsPodReady

Reply:
Indeed, inlined this function.